core/domain/exp_domain.py

```diff
@@ -1298,6 +1316,26 @@ def is_demo(self):
         """
         return self.is_demo_exploration_id(self.id)
 
+    @property
+    def is_stale(self):
+        """Whether the exploration proto size is up to date.
+
+        Returns:
+            int. Updated exploration's proto size.
+        """
+        if self.proto_size_is_stale:
+            self._cached_android_proto_size_in_bytes = self.get_proto_size()
+            self.proto_size_is_stale = False
+
+        return self._cached_android_proto_size_in_bytes
+
+    def __setattr__(self, attrname, new_value):
+        if attrname == 'proto_size_is_stale' and new_value is True:
```

Comment:
This seems completely wrong to me -- you'll want to update self.proto_size_is_stale when most updates occur. The exception is when proto_size_is_stale is set, so that you don't get into an infinite loop.

You do not have to update self.android_proto_size_in_bytes here. That only gets updated when the value is read (see above). The whole point of overriding `__setattr__` is to set the stale flag when a property of the Exploration object is updated.

Also, this method is still missing a docstring with args/returns/etc.

Reply:
> you'll want to update self.proto_size_is_stale when most updates occur

What does most update mean here? Isn't it the same that, for example - the "title or whatever is updated other than `android_proto_size_is_stale` or `android_proto_size_in_bytes` or `_cached_android_proto_size_in_bytes`, let's set the flag true."

> The exception is when proto_size_is_stale is set, so that you don't get into an infinite loop.

Putting a check here will prevent this issue.
```py
    def __setattr__(self, attrname, new_value):
        if (attrname != '_cached_android_proto_size_in_bytes' and
            attrname != 'proto_size_in_bytes' and
            attrname != 'android_proto_size_is_stale'):
            self.android_proto_size_is_stale = True
        super(Exploration, self).__setattr__(attrname, new_value)
```

Reply:
You will never update android_proto_size_in_bytes. See comment above.

And yes I think omitting those two seems correct. Just double-check if there are other hidden member properties of the class that we might need to exclude as well, besides the ones we declare. (I'm not sure if Python sets anything by default on the class.)

The code snippet you've provided seems about right, subject to the caveats above.